### ISSUE TITLE
Mailu deploy openly relaying non-authenticated email

### ISSUE THREAD

**Issue by fiskhest** [Mailu/Mailu#1099]
After successfully running mailu for a couple of weeks a botnet detected that my stack was openly relaying any email thrown at it. The server is behind a router (asus AC68u with asuswrt-merlin firmware), thus traffic coming from the internet is NATed to the internal server. I've run a custom setup postfix+dovecot instance previously under this network configuration, so therefore I think that the network translations should be going correctly, but I haven't gotten the time(/figured out) how to verify that yet. (Would it be wise to tcpdump this?)

When I test against https://www.appmaildev.com/ everything passes except for RBL. (Sorry, some bug with my screenshot taking application so I can't attach an image)

Running swaks yields authentication required:
```
swaks -server domain.tld:587 -4 -tls --to user@gmail.com --from fiskhest@domain.tld
=== Trying domain.tld:587...
=== Connected to domain.tld.
<-  220 domain.tld ESMTP ready
 -> EHLO fisk.hest
<-  250-domain.tld
<-  250 STARTTLS
 -> STARTTLS
<-  220 2.0.0 Start TLS
=== TLS started with cipher TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256
=== TLS no local certificate set
=== TLS peer DN="/CN=domain.tld"
 ~> EHLO fisk.hest
<~  250-domain.tld
<~  250 AUTH PLAIN
 ~> MAIL FROM:<fiskhest@domain.tld>
<~* 530 5.7.1 Authentication required
 ~> QUIT
<~  221 2.0.0 Bye
=== Connection closed with remote host.
```

Testing against https://mxtoolbox.com/diagnostic.aspx:
```
220 domain.tld ESMTP ready [708 ms]
EHLO keeper-us-east-1c.mxtoolbox.com
250-domain.tld
250 STARTTLS [755 ms]
MAIL FROM:<supertool@mxtoolbox.com>
250 2.0.0 OK [725 ms]
RCPT TO:<test@mxtoolboxsmtpdiag.com>
250 2.1.5 Ok [848 ms]
```

My configuration files,
```
# docker-compose.yml
# This file is auto-generated by the Mailu configuration wizard.
# Please read the documentation before attempting any change.
# Generated for compose flavor

version: '3.5'

services:

  # External dependencies
  redis:
    image: redis:alpine
    restart: always
    volumes:
      - "/mailu/redis:/data"
    
  # Core services
  front:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}nginx:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    logging:
      driver: json-file
    ports:
      - "192.168.2.x:8080:80"
      - "192.168.2.x:8443:443"
      - "192.168.2.x:25:25"
      - "192.168.2.x:465:465"
      - "192.168.2.x:587:587"
      - "192.168.2.x:110:110"
      - "192.168.2.x:995:995"
      - "192.168.2.x:143:143"
      - "192.168.2.x:993:993"
    volumes:
    #  - "/mailu/certs:/certs"
    #  - "/etc/letsencrypt:/certs/letsencrypt"
      - "/mailu/overrides/nginx:/overrides"
      - type: bind
        source: /etc/letsencrypt/live/domain.tld/cert.pem
        target: /certs/cert.pem
        read_only: true
      - type: bind
        source: /etc/letsencrypt/live/domain.tld/privkey.pem
        target: /certs/key.pem
        read_only: true


  resolver:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}unbound:${MAILU_VERSION:-1.6}
    env_file: mailu.env
    restart: always
    networks:
      default:
        ipv4_address: 192.168.203.254

  admin:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}admin:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    volumes:
      - "/mailu/data:/data"
      - "/mailu/dkim:/dkim"
    depends_on:
      - redis

  imap:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}dovecot:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    volumes:
      - "/mailu/mail:/mail"
      - "/mailu/overrides:/overrides"
    depends_on:
      - front

  smtp:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}postfix:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    volumes:
      - "/mailu/overrides:/overrides"
    depends_on:
      - front
      - resolver
    dns:
      - 192.168.203.254

  antispam:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}rspamd:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    volumes:
      - "/mailu/filter:/var/lib/rspamd"
      - "/mailu/dkim:/dkim"
      - "/mailu/overrides/rspamd:/etc/rspamd/override.d"
    depends_on:
      - front
      - resolver
    dns:
      - 192.168.203.254

  # Optional services
  antivirus:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}clamav:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    volumes:
      - "/mailu/filter:/data"
    depends_on:
      - resolver
    dns:
      - 192.168.203.254

  webdav:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}radicale:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    volumes:
      - "/mailu/dav:/data"

  fetchmail:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}fetchmail:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    depends_on:
      - resolver
    dns:
      - 192.168.203.254

  # Webmail
  webmail:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}roundcube:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    volumes:
      - "/mailu/webmail:/data"
    depends_on:
      - imap


networks:
  default:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: 192.168.203.0/24
```

```
# mailu.env
# Mailu main configuration file
#
# Generated for compose flavor
#
# This file is autogenerated by the configuration management wizard.
# For a detailed list of configuration variables, see the documentation at
# https://mailu.io

###################################
# Common configuration variables
###################################

# Set this to the path where Mailu data and configuration is stored
# This variable is now set directly in `docker-compose.yml by the setup utility
# ROOT=/mailu

# Mailu version to run (1.0, 1.1, etc. or master)
#VERSION=1.6

# Set to a randomly generated 16 bytes string
SECRET_KEY=<redacted>

# Address where listening ports should bind
# This variables are now set directly in `docker-compose.yml by the setup utility
# PUBLIC_IPV4= 192.168.2.246 (default: 127.0.0.1)
# PUBLIC_IPV6= fe80::9da:5306:2520:edb9 (default: ::1)

# Subnet of the docker network. This should not conflict with any networks to which your system is connected. (Internal and external!)
SUBNET=192.168.203.0/24

# Main mail domain
DOMAIN=domain.tld

# Hostnames for this server, separated with comas
HOSTNAMES=domainx.tld,domainy.tld

# Postmaster local part (will append the main mail domain)
POSTMASTER=admin

# Choose how secure connections will behave (value: letsencrypt, cert, notls, mail, mail-letsencrypt)
TLS_FLAVOR=mail

# Authentication rate limit (per source IP address)
AUTH_RATELIMIT=10/minute;1000/hour 

# Opt-out of statistics, replace with "True" to opt out
DISABLE_STATISTICS=False

###################################
# Optional features
###################################

# Expose the admin interface (value: true, false)
ADMIN=true

# Choose which webmail to run if any (values: roundcube, rainloop, none)
WEBMAIL=roundcube

# Dav server implementation (value: radicale, none)
WEBDAV=radicale

# Antivirus solution (value: clamav, none)
#ANTIVIRUS=clamav

#Antispam solution
ANTISPAM=none

###################################
# Mail settings
###################################

# Message size limit in bytes
# Default: accept messages up to 50MB
# Max attachment size will be 33% smaller
MESSAGE_SIZE_LIMIT=133000000

# Networks granted relay permissions
# Use this with care, all hosts in this networks will be able to send mail without authentication!
RELAYNETS=

# Will relay all outgoing mails if configured
RELAYHOST=

# Fetchmail delay
FETCHMAIL_DELAY=600

# Recipient delimiter, character used to delimiter localpart from custom address part
RECIPIENT_DELIMITER=+

# DMARC rua and ruf email
DMARC_RUA=admin
DMARC_RUF=admin

# Welcome email, enable and set a topic and body if you wish to send welcome
# emails to all users.
WELCOME=false
WELCOME_SUBJECT=Welcome to your new email account
WELCOME_BODY=Welcome to your new email account, if you can read this, then it is configured properly!

# Maildir Compression
# choose compression-method, default: none (value: bz2, gz)
COMPRESSION=
# change compression-level, default: 6 (value: 1-9)
COMPRESSION_LEVEL=

###################################
# Web settings
###################################

# Path to redirect / to
WEBROOT_REDIRECT=/webmail

# Path to the admin interface if enabled
WEB_ADMIN=/admin

# Path to the webmail if enabled
WEB_WEBMAIL=/

# Website name
SITENAME=epost

# Linked Website URL
WEBSITE=https://mail.domain.tld



###################################
# Advanced settings
###################################

# Log driver for front service. Possible values:
# json-file (default)
# journald (On systemd platforms, useful for Fail2Ban integration)
# syslog (Non systemd platforms, Fail2Ban integration. Disables `docker-compose log` for front!)
# LOG_DRIVER=json-file

# Docker-compose project name, this will prepended to containers names.
COMPOSE_PROJECT_NAME=mailu

# Default password scheme used for newly created accounts and changed passwords
# (value: BLF-CRYPT, SHA512-CRYPT, SHA256-CRYPT, MD5-CRYPT, CRYPT)
PASSWORD_SCHEME=BLF-CRYPT

# Header to take the real ip from
REAL_IP_HEADER=

# IPs for nginx set_real_ip_from (CIDR list separated by commas)
REAL_IP_FROM=

# choose wether mailu bounces (no) or rejects (yes) mail when recipient is unknown (value: yes, no)
REJECT_UNLISTED_RECIPIENT=

# Log level threshold in start.py (value: CRITICAL, ERROR, WARNING, INFO, DEBUG, NOTSET)
LOG_LEVEL=WARNING

###################################
# Database settings
###################################
DB_FLAVOR=mysql
DB_USER=username
DB_PW=password
DB_HOST=192.168.203.1
DB_NAME=mailu
```

There are some tweaks I've had to do as I'm running an apache front reverse proxying on the host as I have some other legacy stuff I had to quickly migrate over, it should only be load balancing for the roundcube instance and all other changes shouldn't affect the stack in such a way that it should be open to relaying external networks but I figured it's worth mentioning in case there's something I'm misunderstanding wrt this.

I've got no problem deep diving into containers and network troubleshooting but as I'm a huge jack of all trades I've run out of ideas where to look. Tips are greatly appreciated, thanks.

**Comment by fiskhest** [comment]
So it seems like it's back!

One possible issue mentioned on the chat yesterday was NAT masquerading. Here's a snippet from the source of a relayed email;
```
Received: from kindarium.ru (unknown [192.168.203.1])
by domain.tld (Postfix) with SMTP id 58F5E90F3A61;
```

**Comment by fiskhest** [comment]
After a long time of pondering what could be wrong I finally found that firewalld(on Centos7 atleast) and Docker does not play nice with each other, as per https://github.com/moby/moby/issues/16137

After disabling firewalld (`systemctl stop firewalld && systemctl disable firewalld`) everything magically started working and packets are routed from my NAT gateway to the docker host, through front (and rejected if not authenticated) to smtp-instance. Testing against mxtoolbox yields a closed relay.

Closing issue.
